### ISSUE TITLE
Manjaro-tools 0.9.7-1 possible bug .

### ISSUE THREAD

**Issue by Ste74** [manjaro/manjaro-tools#76]
Hi , i downloaded the new milestone both from our repo that the git-devel but the iso is created not generating errors but when I launch the livecd vbox in the message missing manjaro.img ..
![pearos](https://cloud.githubusercontent.com/assets/11218234/6652811/5324dd8c-ca7c-11e4-8da3-310ac63f2346.png)

**Comment by Kirek** [comment]
A link to the profile of this iso would be helpful

**Comment by Ste74** [comment]
sorry i am new 
https://mega.co.nz/#F!sF8mASgK!aQBucCjeLPjjzE3TfIZFdw

**Comment by Kirek** [comment]
Don't see anything wrong with your profile. 
Can't you try building it with latest devel? The PKGBUILD is here:
https://raw.githubusercontent.com/manjaro/packages-openrc/master/manjaro-tools-git-devel/PKGBUILD

**Comment by Ste74** [comment]
Infact i not understand ..I have build with this 
https://mega.co.nz/#!wF9A2TrD!Qsg-En2i2b-RwWK749osv8uaou-YBjJAUZiP7t5nRsk
I also changed this folder just for completeness 
https://mega.co.nz/#F!EF8TWZCS!90pXqAUycsQrRVv3EtUvsA

**Comment by Kirek** [comment]
In profile.conf you have
"plymouth_theme=plymouth-theme-pearosx-redefined-bsplash"

Those "plymouth-theme-pearosx-redefined-bsplash" match the dir of that theme in "/usr/share/plymouth/themes/" ?

**Comment by Ste74** [comment]
Sorry Kirek .. how i see this ? i look in /opt/buildiso/cinnamon/ecc ecc but i not see

**Comment by Kirek** [comment]
/opt/buildiso/cinnamon/x86_64/cinnamon-image/usr/share/plymouth/themes/
should be there

**Comment by Ste74** [comment]
Aargh is different .. is an error ?
![nuovo4](https://cloud.githubusercontent.com/assets/11218234/6653179/6dcd3782-ca87-11e4-9f20-71e27c91ad86.png)

**Comment by Kirek** [comment]
yes, should be plymouth_theme=Pearosx-Redefined-Bsplash

**Comment by Ste74** [comment]
I try it ... are a rookie 40 years trying to learn as much as possible ..

i modified this and try to rebuild    :-)

**Comment by Kirek** [comment]
Let's hope this solves it.
Also check after building the iso that manjaro.img exist in:
/opt/buildiso/cinnamon/x86_64/iso/manjaro/boot/x86_64/

**Comment by Ste74** [comment]
Ok thank you so much

**Comment by Ste74** [comment]
![nuovo5](https://cloud.githubusercontent.com/assets/11218234/6653393/1958f6f0-ca91-11e4-9c31-a22ddd6bddbf.png)
Ahia  :-( 
manjaro.img not exist

**Comment by philmmanjaro** [comment]
Well, you should check our profile changes in **isolinux.cfg** also. Our current stack looks like this:
<pre>label start
  kernel /%ISO_NAME%/boot/%ARCH%/%ISO_NAME%
  append initrd=/%ISO_NAME%/boot/intel_ucode.img,/%ISO_NAME%/boot/%ARCH%/%ISO_NAME%.img misobasedir=%ISO_NAME% misolabel=%MISO_LABEL% nouveau.modeset=1 i915.modeset=1 radeon.modeset=1 logo.nologo overlay=free quiet splash showopts

label nonfree
  kernel /%ISO_NAME%/boot/%ARCH%/%ISO_NAME%
  append initrd=/%ISO_NAME%/boot/intel_ucode.img,/%ISO_NAME%/boot/%ARCH%/%ISO_NAME%.img misobasedir=%ISO_NAME% misolabel=%MISO_LABEL% nouveau.modeset=0 i915.modeset=1 radeon.modeset=0 nonfree=yes logo.nologo overlay=nonfree quiet splash showopts</pre>

Yours looks like this:
<pre>label start
  kernel /%INSTALL_DIR%/boot/%ARCH%/manjaroiso
  append initrd=/%INSTALL_DIR%/boot/%ARCH%/manjaro.img misobasedir=%INSTALL_DIR% misolabel=%MISO_LABEL% nouveau.modeset=1 i915.modeset=1 radeon.modeset=1 logo.nologo overlay=free showopts
  
label nonfree
  kernel /%INSTALL_DIR%/boot/%ARCH%/manjaroiso
  append initrd=/%INSTALL_DIR%/boot/%ARCH%/manjaro.img misobasedir=%INSTALL_DIR% misolabel=%MISO_LABEL% nouveau.modeset=0 i915.modeset=1 radeon.modeset=0 nonfree=yes logo.nologo overlay=nonfree showopts</pre>

You should have intel-ucode in your packages list also. This is how our default boot folder looks like:
<pre>[phil@manjaro iso]$ tree .
.
├── EFI
│   ├── boot
│   │   ├── bootx64.efi
│   │   ├── HashTool.efi
│   │   └── loader.efi
│   ├── miso
│   │   └── manjaro.img
│   ├── shellx64_v1.efi
│   └── shellx64_v2.efi
├── isolinux
│   ├── 16x16.fnt
│   ├── about.txt.disabled
│   ├── af.tr
│   ├── ar.tr
│   ├── back640x480.jpg
│   ├── back800x600.jpg
│   ├── bg.tr
│   ├── boot.cat
│   ├── bootlogo
│   ├── ca.hlp
│   ├── ca.tr
│   ├── ca@valencia.tr
│   ├── cd_a.jpg
│   ├── cd.jpg
│   ├── chain.c32
│   ├── cs.tr
│   ├── da.tr
│   ├── de.tr
│   ├── el.tr
│   ├── en.hlp
│   ├── en.tr
│   ├── eo.tr
│   ├── es.tr
│   ├── et.tr
│   ├── fi.tr
│   ├── fr.tr
│   ├── gfxboot.c32
│   ├── gfxboot.cfg
│   ├── gl.tr
│   ├── gu.tr
│   ├── hdt.c32
│   ├── hi.tr
│   ├── hr.tr
│   ├── hu.tr
│   ├── id.tr
│   ├── init
│   ├── isohdpfx.bin
│   ├── isolinux.bin
│   ├── isolinux.cfg
│   ├── isolinux.msg
│   ├── it.tr
│   ├── ja.tr
│   ├── ka.tr
│   ├── ko.tr
│   ├── ky.tr
│   ├── languages
│   ├── ldlinux.c32
│   ├── libcom32.c32
│   ├── libgpl.c32
│   ├── libmenu.c32
│   ├── libutil.c32
│   ├── lt.tr
│   ├── mboot.c32
│   ├── memtest
│   ├── mr.tr
│   ├── nb.tr
│   ├── nl.tr
│   ├── pa.tr
│   ├── pci.ids
│   ├── pl.tr
│   ├── pt_BR.tr
│   ├── pt.tr
│   ├── ro.tr
│   ├── ru.tr
│   ├── sk.tr
│   ├── sl.tr
│   ├── sr@latin.tr
│   ├── sr.tr
│   ├── sv.tr
│   ├── ta.tr
│   ├── text.jpg
│   ├── tg.tr
│   ├── th.tr
│   ├── timer_a.jpg
│   ├── tr.tr
│   ├── uk.tr
│   ├── wa.tr
│   ├── welcome.jpg
│   ├── whichsys.c32
│   ├── xh.tr
│   ├── zh_CN.tr
│   ├── zh_TW.tr
│   └── zu.tr
├── loader
│   ├── entries
│   │   ├── manjaro-x86_64.conf
│   │   ├── manjaro-x86_64-nonfree.conf
│   │   ├── uefi-shell-v1-x86_64.conf
│   │   └── uefi-shell-v2-x86_64.conf
│   └── loader.conf
└── manjaro
    ├── boot
    │   ├── intel_ucode.img
    │   ├── intel_ucode.LICENSE
    │   └── x86_64
    │       ├── manjaro
    │       ├── manjaro.img
    │       └── memtest
    ├── isomounts
    └── x86_64
        ├── kde-image.sqfs
        ├── livecd-image.sqfs
        ├── pkgs-image.sqfs
        └── root-image.sqfs

10 directories, 104 files
</pre>

**Comment by philmmanjaro** [comment]
Also provide us with your **build.log**. A normal **manjaro.img** generation looks like this:
<pre>==> Prepare [manjaro/boot]
  -> mount [boot-image] on [xfce-image]
  -> append [root-image] on [boot-image]
  -> Copying initcpio ...
==> Starting build: 3.19.1-1-MANJARO
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [plymouth]
  -> Running build hook: [miso]
  -> Running build hook: [miso_pxe_nbd]
  -> Running build hook: [miso_loop_mnt]
  -> Running build hook: [miso_kms]
  -> Running build hook: [modconf]
  -> Running build hook: [block]
==> WARNING: Possibly missing firmware for module: wd719x
==> WARNING: Possibly missing firmware for module: aic94xx
  -> Running build hook: [pcmcia]
  -> Running build hook: [filesystems]
  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
==> Generating module dependencies
==> Creating xz-compressed initcpio image: /boot/manjaro.img
==> Image generation successful
  -> unmount boot-image
==> Done [manjaro/boot]</pre>

**Comment by Ste74** [comment]
Thank you Philip for answer ... Where is the build.log for verify

**Comment by philmmanjaro** [comment]
When you run **buildiso** you have to pass the **-L** switch to generate one. Else check the termial output. Therefore please adjust your terminal to keep at least 5000 lines in scroll-back.

**Comment by Ste74** [comment]
Ok , run a new build

**Comment by Ste74** [comment]
-> Computing checksums...
  -> Creating 'desc' db entry...
  -> Creating 'depends' db entry...
==> Adding package '/opt/buildiso/cinnamon/x86_64/pkgs-image/opt/livecd/pkgs/xf86-video-voodoo-1.2.5-6-x86_64.pkg.tar.xz'
  -> Computing checksums...
  -> Creating 'desc' db entry...
  -> Creating 'depends' db entry...
==> Creating updated database file '/opt/buildiso/cinnamon/x86_64/pkgs-image/opt/livecd/pkgs/gfx-pkgs.db.tar.gz'
  -> unmount pkgs-image
==> Done [pkgs-image]
==> Prepare [manjaro/boot]
  -> mount [boot-image] on [cinnamon-image]
  -> append [root-image] on [boot-image]
  -> Copying initcpio ...
==> ERROR: '/lib/modules/3.18.9-1-MANJARO' is not a valid kernel module directory
mv: cannot stat '/opt/buildiso/cinnamon/x86_64/boot-image/boot/manjaro.img': No such file or directory
  -> unmount boot-image
==> Done [manjaro/boot]
==> Prepare [manjaro/boot/EFI]
  -> Copying efi loaders ...
  -> Writing loader.conf ...
  -> Writing uefi-shell-v1-x86_64.conf ...
  -> Writing uefi-shell-v2-x86_64.conf ...
  -> Writing manjaro-x86_64.conf ...
  -> Writing manjaro-x86_64-nonfree.conf ...
  -> Copying shellx64_v1.efi ...
  -> Copying shellx64_v2.efi ...
==> Done [manjaro/boot/EFI]
==> Prepare [manjaro/iso/EFI]
mkfs.fat 3.0.27 (2014-11-12)
  -> Copying boot images ...
cp: cannot stat '/opt/buildiso/cinnamon/x86_64/iso/manjaro/boot/x86_64/manjaro.img': No such file or directory
  -> Copying efi loaders ...
  -> Writing loader.conf ...
  -> Writing uefi-shell-v1-x86_64.conf ...
  -> Writing uefi-shell-v2-x86_64.conf ...
  -> Writing manjaro-x86_64.conf ...
  -> Writing manjaro-x86_64-nonfree.conf ...
  -> Copying shellx64_v1.efi ...
  -> Copying shellx64_v2.efi ...
==> Done [manjaro/iso/EFI]
==> Prepare [manjaro/iso/isolinux]
  -> Writing isolinux.cfg ...
  -> Writing isolinux.msg ...
  -> Copying isolinux bios binaries ...
==> Done [manjaro/iso/isolinux]
==> Creating [isomounts]
  -> Writing livecd entry ...
  -> Writing pkgs entry ...
  -> Writing cinnamon entry ...
  -> Writing root entry ...
==> Done creating [isomounts]
 --> Time build_images: 9.32 minutes
==> WARNING: Continue compress: buildiso -p cinnamon -sc ...
[ste74@Hp iso-profiles]$ 

Sorry for paste the lines .. so i see an error

**Comment by philmmanjaro** [comment]
So your issue is here:
<pre>==> Prepare [manjaro/boot]
-> mount [boot-image] on [cinnamon-image]
-> append [root-image] on [boot-image]
-> Copying initcpio ...
==> ERROR: '/lib/modules/3.18.9-1-MANJARO' is not a valid kernel module directory
mv: cannot stat '/opt/buildiso/cinnamon/x86_64/boot-image/boot/manjaro.img': No such file or directory
-> unmount boot-image
==> Done [manjaro/boot]</pre>

This tells me that some is off. I'll check the coping step now ...

**Comment by Ste74** [comment]
Glad to be of help

**Comment by philmmanjaro** [comment]
I think this step you pass ...
<pre>copy_initcpio(){
	msg2 "Copying initcpio ..."
	cp /usr/lib/initcpio/hooks/miso* $1/usr/lib/initcpio/hooks
	cp /usr/lib/initcpio/install/miso* $1/usr/lib/initcpio/install
	cp mkinitcpio.conf $1/etc/mkinitcpio-${iso_name}.conf
}</pre>

But you fail on this one:
<pre>gen_boot_image(){
	chroot-run $1 \
		/usr/bin/mkinitcpio -k $(uname -r) \
		-c /etc/mkinitcpio-${iso_name}.conf \
		-g /boot/${iso_name}.img
}</pre>

This means you have to get the kernel right. Please validate your [profile.conf](https://github.com/manjaro/manjaro-tools-iso-profiles/raw/master/cinnamon/profile.conf) and [mkinitcpio.conf](https://github.com/manjaro/manjaro-tools-iso-profiles/raw/master/shared/mkinitcpio.conf). Did you ever tried to build **kde5**, **xfce** or **xfce-minimal**? Those profiles are tested. Also you can't just copy **manjaroiso** profiles 1:1 to **manjaro-tools**. Your issues lay within your profile. Please upload it somewhere so we can verify it. Or you check our **cinnamon** profile and fix yours.

**Comment by Ste74** [comment]
No problem for me to share my profile cinnamon i upload the cinnamon folder in iso profile and the shared folder right ?

**Comment by philmmanjaro** [comment]
Well, simply fork [manjaro-tools-iso-profiles](https://github.com/manjaro/manjaro-tools-iso-profiles), add your changes to it and grand me access. Then I might be able to help you out. Please use the devel branch.

**Comment by Ste74** [comment]
Ok thank again Philip

**Comment by philmmanjaro** [comment]
To work with **manjaro-tools** either use the given binary packages we release or switch over to the development tools. Therefore we have **PKGBUILD**s. Do followed steps to get there:

<pre>wget https://github.com/manjaro/packages-openrc/raw/master/manjaro-tools-git-devel/PKGBUILD
makepkg -i
git clone https://github.com/manjaro/manjaro-tools-iso-profiles.git
cd manjaro-tools-iso-profiles
git checkout devel</pre>

Then try to build **xfce-minimal**:
<pre>buildiso -q
sudo buildiso</pre>

**Comment by Ste74** [comment]
ok i follow the steps . I buildiso xfce-minimal .
I also have forked profiles . How to upload my change later ?

**Comment by Ste74** [comment]
![nuovo6](https://cloud.githubusercontent.com/assets/11218234/6655098/6ab5a68a-caee-11e4-8ded-bd3bdf5ce65f.png)
Same error with your step for xfce-minimal

**Comment by philmmanjaro** [comment]
Do you really have the latest **manjaro-tools.conf** in your **~/.config** folder? Mine would be this one here:
<pre>######################################################
################ manjaro-tools.conf ##################
######################################################

# unset defaults to given value
branch=unstable

# unset defaults to given value
# arch=$(uname -m)

# cache dir where buildpkg or buildiso cache packages
# cache_dir=/var/cache/manjaro-tools

################ buildtree ###############

# manjaro package tree
# repo_tree=('core' 'extra' 'community' 'multilib' 'openrc')

# host_tree=https://github.com/manjaro

# host_tree_abs=https://projects.archlinux.org/git/svntogit

################ buildpkg ################

# default chroot path
# chroots_pkg=/opt/buildpkg

# custom path to pkg sets
# sets_dir_pkg=/etc/manjaro-tools/sets/pkg

# default pkg buildset; name without .set extension
# buildset_pkg=default

# custom build mirror server
# build_mirror=http://mirror.netzspielplatz.de/manjaro/packages

# Next settings are only useful if you compile packages against eudev

# default packages to trigger blacklist
# blacklist_trigger=('eudev' 'upower-pm-utils' 'eudev-systemdcompat')

# default blacklisted packages to remove from chroot
# blacklist=('libsystemd')

################ buildiso ################

# default work dir where the image chroots are located
# chroots_iso=/opt/buildiso

# custom path to iso sets
# sets_dir_iso=/etc/manjaro-tools/sets/iso

# default iso buildset; name without .set extension
# buildset_iso=default

# unset defaults to given value
# dist_name="Manjaro"

# unset defaults to given value
dist_release=0.9.0-dev

# unset defaults to value sourced from /etc/lsb-release
dist_codename="Bellatrix"

# unset defaults to given value
# dist_branding="MJRO"

# unset defaults to given value, specify a date here of have it automatically set
# dist_version="$(date +%Y.%m)"

# unset defaults to given value
# iso_name=manjaro

# unset defaults to given value
# iso_compression=xz

# valid: md5, sha1, sha256, sha384, sha512
# iso_checksum=md5</pre>

**Comment by philmmanjaro** [comment]
Also pastebin your complete build log, so I can see the step you're hanging:
<pre>buildiso -q
sudo buildiso</pre>
Here are mine:
[manjaro-xfce-0.9.0-dev-i686-build.log](http://sourceforge.net/projects/manjarotest/files/0.9.0/xfce-minimal/0.9.0-dev/manjaro-xfce-0.9.0-dev-i686-build.log)
[manjaro-xfce-0.9.0-dev-x86_64-build.log](http://sourceforge.net/projects/manjarotest/files/0.9.0/xfce-minimal/0.9.0-dev/manjaro-xfce-0.9.0-dev-x86_64-build.log)

**Comment by philmmanjaro** [comment]
Ok, I was able now to reproduce your issue:
<pre>==> Prepare [manjaro/boot]
  -> mount [boot-image] on [xfce-image]
  -> append [root-image] on [boot-image]
  -> Copying initcpio ...
==> ERROR: '/lib/modules/3.18.9-1-MANJARO' is not a valid kernel module directory
mv: cannot stat '/opt/buildiso/xfce-minimal/x86_64/boot-image/boot/manjaro.img': No such file or directory
  -> unmount boot-image
</pre>

I had followed settings on my host:
<pre>[phil@manjaro manjaro-tools-iso-profiles]$ pacman -Q | grep manjaro-tools
manjaro-tools-git r946.13a8baf-1
[phil@manjaro manjaro-tools-iso-profiles]$ uname -a
Linux manjaro 3.18.9-1-MANJARO #1 SMP PREEMPT Sat Mar 7 18:16:39 UTC 2015 x86_64 GNU/Linux
[phil@manjaro manjaro-tools-iso-profiles]$ inxi -F
System:    Host: manjaro Kernel: 3.18.9-1-MANJARO x86_64 (64 bit)
           Desktop: Xfce 4.12.0-MANJARO Distro: ManjaroLinux 0.8.12 Ascella
[phil@manjaro manjaro-tools-iso-profiles]$ buildiso -q
==> manjaro-tools
  -> version: 0.9.7
  -> user_config: /home/phil/.config/manjaro-tools.conf
==> PROFILE:
  -> buildsets: community|default|official|openrc
  -> buildset_iso: default
  -> is_buildset: true
==> OPTIONS:
  -> arch: x86_64
  -> branch: unstable
  -> chroots_iso: /opt/buildiso
==> ARGS:
  -> clean_first: true
  -> clean_cache_xorg: true
  -> clean_cache_lng: true
  -> clean_cache_iso: true
  -> images_only: false
  -> iso_only: false
==> PATHS:
  -> sets_dir_iso: /etc/manjaro-tools/sets/iso
  -> cache_dir_iso: /var/cache/manjaro-tools/iso
  -> cache_dir_xorg: /var/cache/manjaro-tools/xorg/x86_64
  -> cache_dir_lng: /var/cache/manjaro-tools/lng/x86_64
==> DIST SETTINGS:
  -> dist_name: Manjaro
  -> dist_release: 0.9.0-dev
  -> dist_codename: Bellatrix
  -> dist_version: 2015.03
==> ISO SETTINGS:
  -> iso_label: MJRO090D
  -> iso_name: manjaro
  -> iso_compression: xz
  -> iso_checksum: md5
==> BUILD QUEUE:
 --> Profile: [xfce-minimal]
  -> Packages-Custom: Packages-Xfce
  -> work_dir: /opt/buildiso/xfce-minimal/x86_64
  -> iso_file: manjaro-xfce-0.9.0-dev-x86_64.iso
  -> pacman_conf: /usr/share/manjaro-tools/pacman-multilib.conf
  -> initsys: systemd
  -> displaymanager: lightdm
  -> kernel: linux319
  -> efi_boot_loader: grub
  -> plymouth_theme: manjaro-elegant
  -> hostname: manjaro
  -> username: manjaro
  -> password: manjaro
  -> addgroups: video,audio,power,disk,storage,optical,network,lp,scanner,wheel
  -> start_systemd: bluetooth cronie ModemManager NetworkManager org.cups.cupsd tlp tlp-sleep
  -> start_systemd_live: livecd mhwd-live pacman-init pacman-boot
</pre>

In short: mkinitcpio uses the modules folder from your host kernel. I didn't noticed this issue as I was using linux319 as my host-kernel. I'm working now on a fix.

**Comment by philmmanjaro** [comment]
Your issues is fixed with 4a5f96d54f73b47e7b436554c05c253554467987. Please do this:
<pre>wget https://github.com/manjaro/packages-openrc/raw/master/manjaro-tools-git-devel/PKGBUILD
makepkg -i</pre>

**Comment by Ste74** [comment]
Thank you Philip , today is sunday but you are at work ...
Yes even I had thought that the error was the kernel module of my system is not updated to the series 3.19.
Thanks again try my profile with the latest PKGBUILD. :-)

**Comment by philmmanjaro** [comment]
At work I've other things to do and have during the week not so much time for Manjaro. I think it should work now. I also released **0.9.7.1**, which includes this fix.

**Comment by udeved** [comment]
Can we close this?

Is it still issue with 0.9.7.4?

**Comment by Kirek** [comment]
It's solved you can close it.

**Comment by Ste74** [comment]
Sorry, I had forgotten about that .. I thought it was closed ...

**Comment by philmmanjaro** [comment]
@Ste74: with your answer you reopend it last time ... ;)
